### PR TITLE
[Cleanup]: Fix Android Studio Warnings in various Kotlin files #13282

### PR DESCRIPTION
**Pull request by BAH-HA**
## Purpose / Description
Cleaning Android Studio Warnings in various Kotlin files
## Fixes
* Related to #13282 

## Approach
Removed unused imports, changed some java functions to kotlin ones and fixed some code style issues

## How Has This Been Tested?

Ran ./gradlew jacocoUnitTestReport and ./gradlew jacocoAndroidTestReport

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [ X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ X] You have commented your code, particularly in hard-to-understand areas
- [X ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by david-allison — APPROVED** [review]
Perfect, cheers!

In future, look into `squash`, `rebase` + `force-push` so we have a linear git history when merging

**Comment by BAH-HA** [comment]
> Perfect, cheers!
> 
> In future, look into `squash`, `rebase` + `force-push` so we have a linear git history when merging.
> 
> Ideally each pull request should be available for us to "Rebase Merge" into main.

Thank you for the Feedback David, I'll take that into consideration in my future PR.

File: api/src/main/java/com/ichi2/anki/api/AddContentApi.kt [hunk 1]
```diff
@@ -36,6 +35,7 @@ import com.ichi2.anki.FlashCardsContract.Model
 import com.ichi2.anki.FlashCardsContract.Note
 import java.io.File
 import java.util.*
+import com.ichi2.anki.FlashCardsContract as FlashCardsContract
```
Comment (lukstbit) [inline api/src/main/java/com/ichi2/anki/api/AddContentApi.kt:38]: There isn't any issue highlighted by the IDE for this import and the change itself doesn't fix anything. Please revert this.

File: AnkiDroid/src/test/java/com/ichi2/anki/servicelayer/scopedstorage/OperationTest.kt [hunk 2]
```diff
@@ -97,7 +97,7 @@ interface OperationTest {
          * The [MoveDirectoryContent] that performs the action mentioned in the class description.
          */
         val spy: MoveDirectoryContent
-            get() = spy(moveDirectoryContent) {
+            get() = spy(moveDirectoryContent) { it ->
```
Comment (lukstbit) [inline AnkiDroid/src/test/java/com/ichi2/anki/servicelayer/scopedstorage/OperationTest.kt:100]: Using _it_ for the lambda parameter name is confusing as the same name is used below with two different values + the standard name for the parameter of a lambda being _it_.
Please rename it to _moveDirectoryContent_ and change its usage below in _whenever_ as well.

**Review by lukstbit — CHANGES_REQUESTED** [review]
Thanks for contributing!
I left two issues to fix and then we could merge this PR.

File: api/src/main/java/com/ichi2/anki/api/AddContentApi.kt (hunk 1 above)
Comment (BAH-HA) [reply api/src/main/java/com/ichi2/anki/api/AddContentApi.kt]: Thank you for your Feedback, already committed the new changes

File: AnkiDroid/src/test/java/com/ichi2/anki/servicelayer/scopedstorage/OperationTest.kt (hunk 2 above)
Comment (BAH-HA) [reply AnkiDroid/src/test/java/com/ichi2/anki/servicelayer/scopedstorage/OperationTest.kt]: Done 👍

**Review by lukstbit — APPROVED** [review]
All good, thank you for making the changes.

**Pull request merged**